src/osd/OSD.cc

```diff
@@ -9895,13 +9895,10 @@ void OSD::ShardedOpWQ::_process(uint32_t thread_index, heartbeat_handle_d *hb)
   sdata->sdata_op_ordering_lock.Lock();
   if (sdata->pqueue->empty()) {
     dout(20) << __func__ << " empty q, waiting" << dendl;
-    // optimistically sleep a moment; maybe another work item will come along.
-    osd->cct->get_heartbeat_map()->reset_timeout(hb,
-      osd->cct->_conf->threadpool_default_timeout, 0);
+    osd->cct->get_heartbeat_map()->clear_timeout(hb);
```

Comment:
This feels dangerous to me  since it will be equivalent to tossing out the heatlh-checking mechanism. E.g., we'll be unable to detect a bad-behavioured OSD which is currently suffering high disk latency...

Reply:
Yes,you are right. I think we should reset_timeout  in line9908.